pennylane/templates/subroutines/permute.py

```diff
@@ -157,15 +161,62 @@ def __init__(self, permutation, wires, do_queue=True, id=None):
             # Make sure everything in the permutation has an associated label in wires
         for label in permutation:
             if label not in wires:
-                raise ValueError(f"Cannot permute wire {label} not present in wire set.")
+                raise ValueError(
+                    f"Cannot permute wire {label} not present in wire set.")
 
-        self._hyperparameters = {"permutation": permutation}
+        self._hyperparameters = {"permutation": permutation, 'wires': wires}
         super().__init__(wires=wires, do_queue=do_queue, id=id)
 
     @property
     def num_params(self):
         return 0
 
+    def matrix(self):  # pylint: disable=arguments-differ
+
+        wires = self._hyperparameters['wires']
+        permutation = self._hyperparameters['permutation']
+        n_wires = len(wires)
+
+        wires_indices = list(range(n_wires))
+        perm_indices = [wires.index(label) for label in permutation]
```

Comment:
Why not use `wires` and `permutation` directly?

You could loop over the `permutation` values and if the values are different from the ones in `wires` then find the swapping index (see comment below).